test/types.js

```diff
@@ -0,0 +1,79 @@
+import {readdirSync, statSync} from 'fs';
+import {resolve as _resolve, join, extname, relative} from 'path';
+import pkgUp from 'pkg-dir';
+
+import test from 'ava';
+import * as ts from 'typescript';
+
+const resolve = _resolve.bind(null, __dirname);
+const validExtension = new Set(['.js', '.ts']);
+const compilerOptions = {
+	target: ts.ScriptTarget.ES2017,
+	module: ts.ModuleKind.CommonJS,
+	strict: true,
+	noEmit: true
+};
+
+test('TS can compile valid Emittery method calls [slow]', assertAllCompile, './fixtures/compiles');
+test('TS warns about invalid Emittery method calls [slow]', assertEachFail, './fixtures/fails');
+
+function assertAllCompile(t, srcDir) {
+	const fileNames = listFiles(resolve(srcDir));
+	const errors = compile(fileNames);
+
+	t.is(errors.length, 0, errorMessage(errors));
+}
+
+function assertEachFail(t, srcDir) {
+	const fileNames = listFiles(resolve(srcDir)).sort();
+	const errors = compile(fileNames);
+	const filesWithErrors = errors
+		.map(err => (err.file ? err.file.fileName : null))
+		.filter(Boolean)
+		.filter(unique);
```

Comment:
You could pass the boolean-filtered array to `new Set()` to get unique values. If you do the same with `fileNames` you can still use `t.deepEqual()`.